modules/end-user-portal/src/Pages/Search/Results.cshtml

```diff
@@ -19,26 +19,66 @@
                 <div class="nhsuk-card">
                     <div class="nhsuk-card__content">
                         <h3 class="nhsuk-card__heading">
-                            Clinical care setting results for '@Model.Query'
+                            Found @Model.TotalResults clinical care settings mathing '@Model.Query'
                         </h3>
                         <h4>Name and address</h4>
                         <dl class="nhsuk-summary-list">
                             @foreach (var searchResult in Model.SearchResult.SearchResults)
                             {
                                 <div class="nhsuk-summary-list__row">
-                                <dt class="nhsuk-summary-list__key">
-                                   <a asp-page="./Detail"
-                                   asp-route-id="@searchResult.SiteDefinitionId"
-                                   asp-route-query="@Model.Query"
-                                   asp-route-mode="@Model.Mode"
-                                   asp-route-source=@DetailViewSource.Results>@searchResult.FullAddress</a>
-                                </dt>
-                        </div>
+                                    <dt class="nhsuk-summary-list__key">
+                                        <a asp-page="./Detail"
+                                            asp-route-id="@searchResult.SiteDefinitionId"
+                                            asp-route-query="@Model.Query"
+                                            asp-route-mode="@Model.Mode"
+                                            asp-route-source=@DetailViewSource.Results
+                                            asp-route-pageNumber=@Model.PageNumber
+                                        >
+                                                @searchResult.FullAddress
+                                        </a>
+                                    </dt>
+                                </div>
                             }
-                    </dl>
-                </div>
-                <div>
-                </div>
+                        </dl>
+                    
+                        @if (Model.ShowPreviousLink || Model.ShowNextLink) 
+                        {
+                            <nav class="nhsuk-pagination" role="navigation" aria-label="Pagination">
+                                <ul class="nhsuk-list nhsuk-pagination__list">
+                                    @if (Model.ShowPreviousLink) 
+                                    {
+                                        <li class="nhsuk-pagination-item--previous">
+                                            <a class="nhsuk-pagination__link nhsuk-pagination__link--prev" 
```

Comment:
Could put the previous and next markup into a separate single partial view, passing through the necessary parameters, as the markup is virtually identical

Reply:
There are 5 elements that need to be config-ed in a common partial, I think that for now it is simpler in the current format. Happy to discuss about this.